File: algorithms/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        l3 = None
        n1, n2 = l1, l2
        pre = l3
        adv = 0
        while n1 or n2 or adv:
            v1, v2 = n1.val if n1 else 0, n2.val if n2 else 0
            ret = v1 + v2 + adv
            adv = ret // 10
            now = ListNode(x=ret % 10)
            if pre:
                pre.next = now
            else:
                l3 = now
            n1, n2, pre = n1.next if n1 else None, n2.next if n2 else None, now
        return l3

File: algorithms/0002-add-two-numbers.go
package algorithms

func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	var l3 *ListNode;
	var pr *ListNode;
	var ad int = 0;
	for (l1 != nil) || (l2 !=  nil) || (ad != 0) {
		var v1, v2 int;
		if l1 != nil {
			v1 = l1.Val;
		} else {
			v1 = 0;
		}
		if l2 != nil {
			v2 = l2.Val;
		} else {
			v2 = 0;
		}
		var re int = v1 + v2 + ad;
		var ne ListNode;
		ad = re / 10;
		ne.Val = re % 10;
		ne.Next = nil;
		if pr != nil {
			pr.Next = &ne;
		} else {
			l3 = &ne;
			pr = l3;
		}
		if l1 != nil {
			l1 = l1.Next;
		} else {
			l1 = nil;
		}
		if l2 != nil {
			l2 = l2.Next;
		} else {
			l2 = nil;
		}
		pr = &ne;
	}
	return l3;
}
